lib/tokenize.cpp

```diff
@@ -8472,6 +8472,10 @@ void Tokenizer::findGarbageCode() const
             if (Token::Match(tok, "> %cop%"))
                 continue;
         }
+        if (Token::Match(tok, "%or%|%oror%|%%|==|!=|+|-|/|!|>=|<=|~|++|--|::|sizeof|throw|decltype|typeof {"))
```

Comment:
aha!
when I build cppcheck with self-checks (it has some checks that make sure it its own match patterns are well formed)
` make CXXFLAGS="-DCHECK_INTERNAL"`
and use the resulting binary on your code, I get
````
[lib/tokenize.cpp:8475]: (error) Unknown pattern used: "%%"
````
I know we already had such cases where bad patterns would just cause "internal errors" in the past.
When I remove `%%|` from the line, testrunner no longer segfaults :)

Reply:
Thanks, I removed this.

Reply:
Should `Token::Match()` (see http://cppcheck.sourceforge.net/devinfo/doxyoutput/classToken.html#a519623098bde6496825e49364575dadf) be able to handle this?

Reply:
I am not sure but I would expect that `%|` works.